aptos-move/block-executor/src/proptest_types/types.rs

```diff
@@ -235,15 +241,22 @@ impl Default for TransactionGenParams {
     }
 }
 
+// TODO: move generation to separate file.
+// TODO: consider adding writes to reads (read-before-write). Similar behavior to the Move-VM
+// and may force more testing (since we check read results).
 impl<V: Into<Vec<u8>> + Arbitrary + Clone + Debug + Eq + Sync + Send> TransactionGen<V> {
     fn writes_and_deltas_from_gen<K: Clone + Hash + Debug + Eq + Ord>(
```

Comment:
Any reason why we generate writes and deltas in the same function?

Reply:
This is another hack, the way deltas were tested was that some keys would be considered aggregator keys and when a random write was generated there, we would convert it to delta or a write.. (because we still want to test writes to aggregators).
I added a TODO in the code and also in the PR description - will clean transaction generation in a follow-up (even though I also touched it up a little bit now)